From 9a3379a189d3cdeaa37c3887c0891ece5e943061 Mon Sep 17 00:00:00 2001
From: Monk Liu <Monk.Liu@amd.com>
Date: Thu, 6 Apr 2017 15:30:39 +0800
Subject: [PATCH 2700/3082] drm/amdgpu:fix psp ip block patch

should use kernel parm to judge instead of adev member,
which hasn't been set at that time

Change-Id: I58c83a8615cd4a54440c9de038e45bb70802cc2c
Signed-off-by: Monk Liu <Monk.Liu@amd.com>
Reviewed-by: Xiangliang Yu <Xiangliang.Yu@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/soc15.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/soc15.c b/drivers/gpu/drm/amd/amdgpu/soc15.c
index 35db2c7..aba1c16 100644
--- a/drivers/gpu/drm/amd/amdgpu/soc15.c
+++ b/drivers/gpu/drm/amd/amdgpu/soc15.c
@@ -497,7 +497,7 @@ int soc15_set_ip_blocks(struct amdgpu_device *adev)
 		amdgpu_ip_block_add(adev, &mmhub_v1_0_ip_block);
 		amdgpu_ip_block_add(adev, &gmc_v9_0_ip_block);
 		amdgpu_ip_block_add(adev, &vega10_ih_ip_block);
-		if (adev->firmware.load_type == AMDGPU_FW_LOAD_PSP)
+		if (amdgpu_fw_load_type == 2 || amdgpu_fw_load_type == -1)
 			amdgpu_ip_block_add(adev, &psp_v3_1_ip_block);
 		if (!amdgpu_sriov_vf(adev))
 			amdgpu_ip_block_add(adev, &amdgpu_pp_ip_block);
-- 
2.7.4

